### ISSUE TITLE
`ApolloMutationCall.toBuilder().build()` does not return `ApolloMutationCall`

### ISSUE THREAD

**Issue by mateuszkwiecinski** [apollographql/apollo-android#2515]
**Summary**
`ApolloMutationCall.toBuilder().build()` should return `ApolloMutationCall` but returns `ApolloCall`.

**Version**
2.2.3

**Description**
Before https://github.com/apollographql/apollo-android/pull/2434 it was possible to customize `ApolloMutationCall` by calling i.e. 
```
fun mutationCall(): ApolloMutation<TData> = apolloClient.mutate(...).cacheHeaders(...)
```

the new, recommended way is to call
```
fun mutationCall(): ApolloMutation<TData> = apolloClient.mutate(...)..toBuilder().cacheHeaders(...).build()
```
which doesn't compile because `build()` returns `ApolloCall<TData!>`, not `ApolloMutation<TData>`. 

This doesn't apply to `ApolloQueryCall` which returns proper type.

**Comment by martinbonnin** [comment]
This will be fixed with https://github.com/apollographql/apollo-android/pull/2501. In the meantime, you can keep using the deprecated method, which continues to work as before. 
In the long run, we want to encourage builder patterns because they make it explicit that the return value must be used.

**Comment by mateuszkwiecinski** [comment]
Ah, sorry I missed related https://github.com/apollographql/apollo-android/issues/2500#issuecomment-670404695 . 
Thank you for a quick reply!

**Comment by tasomaniac** [comment]
Should we do a release with the fix?

**Comment by martinbonnin** [comment]
Yep, definitely, I'll make one tomorrow.

**Comment by mateuszkwiecinski** [comment]
I confirm that's been fixed in [2.3.0](https://github.com/apollographql/apollo-android/releases/tag/v2.3.0)
Thank you!
